### DISCUSSION TITLE
How to disable "update broker" aka RustDesk auto update?

### DISCUSSION THREAD

**Discussion by DarkLordGMS** [rustdesk/rustdesk#6423] [General]
I don't want to have the UAC showing up. I prefer to update manually.

**Comment by mavorse** [comment]
I believe that auto-update is just a fancy term for allowing software creators to control the software on a user's computer. I strongly disagree with this practice in any case. It means handing over complete control to the software developers for the installation of an unknown program on my computer (unknown because a new version may have new features that I am unaware of and did not agree to, yet it installs automatically). This is a very dangerous behavior that's become popular in this days, for some unknown reasons. I alone should have control over my computer, and no program should be able to force its installation on my private computer.

**Comment by mavorse** [comment]
I'll go further to say that the mechanism of forced updates is similar to the mechanism of today's malware, which involves downloading software from developers' servers without the user's knowledge or consent and installing it without the user's awareness or agreement.

**Comment by pdifeo** [comment]
> I don't want to have the UAC showing up. I prefer to update manually.

It is really annoying to see this dialog every time a remote computer is accessed. I have disabled the automatic update, but the dialog always appears

**Comment by AlmightyJojo** [comment]
Seeing same thing - believe it began with latest client 1.2.3 ?
Rustdesk client is creating + executing C:\Users\%username%\AppData\Local\Temp\RustDesk_update_broker.bat upon unattended connections

It prompts UAC on regular domain user / non admin everytime.  Connection works fine declining execute and then deletes bat file.

Strangely I disabled setting "check for software update on startup" under a domain admin account and its no longer happening.
However when logging into regular users account the  "check for software update on startup" is still enabled yet no longer prompting.

So my take is to disable it has to be toggled off from a pc admin account despite limited user account allowing disable?

In my opinion all Rustdesk server (incoming) settings should be global under program data and locked behind pc admin credentials. 
But it appears both server (incoming) + client (outgoing) settings are under every user profile toml files but are not all in effect -  C:\Users\%username%\AppData\Roaming\RustDesk\config

RustDesk_update_broker.bat contents

        chcp 65001
        taskkill /F /IM RuntimeBroker_rustdesk.exe
        copy /Y "C:\Windows\System32\RuntimeBroker.exe" "C:\Program Files\RustDesk\RuntimeBroker_rustdesk.exe"
    
if exist "C:\Users\%username%\AppData\Local\Temp\RustDesk_update_broker.bat.undone" del /f /q "C:\Users\%username%\AppData\Local\Temp\RustDesk_update_broker.bat.undone"

Ruskdesk server 1.1.9 (docker)
Rustdesk 1.2.3 (client)

**Comment by carpfff** [comment]
Hello,

Is anybody having a fix or workaround for this problem ?

Kind regards.

**Comment by Koruudo** [comment]
I'm having this same issue, for a while my users kept getting the "This app has been blocked by your system administrator" dialog and I could never figure out what was causing it. Now I found it's UAC automatically denying the elevation request.

My end users do not have local admin so this shouldn't be executing in their user's security context.

**Comment by fufesou** [comment]
Sorry for the inconvenience.

The behavior of this operation is to copy a copy of the system's "broker.exe" to the RustDesk directory and rename it to distinguish the process. Also requires "broker.exe" to be the latest version.

https://github.com/rustdesk/rustdesk/blob/4b0e88ce466b22cfda33ba68c6d42231d52fe513/src/platform/windows.rs#L985

It's a safe operations.

This "broker.exe" is required for privacy mode.
In the latest ngihtly, this check is only performed when privacy mode is turned on, instead of being performed on every connection.

**Comment by Koruudo** [comment]
> 
> I'll go further to say that the mechanism of forced updates is similar to the mechanism of today's malware, which involves downloading software from developers' servers without the user's knowledge or consent and installing it without the user's awareness or agreement.

Ehhh, that's a stretch. Malware has malicious intent; this does not.

**Comment by gacpac** [comment]
any chance we get it resolved on its own in a future update? it's kind of annoying

**Reply by fufesou** [reply]
Sorry for the inconvenience.
Please try the nightly build.
